### PR TITLE
Fix CentOS 6, install EPEL before acceptance tests, add templates for Ubuntu 18.

### LINKED ISSUE #36: Debian Stretch
Good day
A management and a template for Debian Stretch is needed.

Kind Regards
Brent Clark

### LINKED ISSUE #57: Acceptance tests don't work for CentOS 6
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.5.1
- Distribution: CentOS 6

## How to reproduce 

Run in your command line:

```bash
PUPPET_INSTALL_TYPE=agent BEAKER_debug=true BEAKER_set=docker/centos-6 bundle exec rake beaker
```

## What are you seeing

Acceptance tests fail for CentOS 6

## What behaviour did you expect instead

The tests should be working 

## Additional info

@traylenator  started working on https://github.com/voxpupuli/puppet-fail2ban/pull/54 but I've discussed the issues with him and will be picking this up.

### PR DESCRIPTION
**Pull request by saibot94**
#### Pull Request (PR) description

Fixed CentOS 6 tests and ugly command line hacks for the RedHat tests. Using beaker to install the needed packages before running the acceptance tests.

Added template files for Ubuntu 18. Unfortunately, we can't enable ubuntu-18.04 tests yet, as the package needed by beaker doesn't exist.

Try running `PUPPET_INSTALL_TYPE=agent BEAKER_debug=true BEAKER_set=docker/ubuntu-18.04 bundle exec rake beaker` and you'll get:

```
ubuntu-1804-x64 executed in 0.48 seconds
Exited: 8

An error occurred while loading ./spec/acceptance/class_spec.rb.
Failure/Error: run_puppet_install_helper unless ENV['BEAKER_provision'] == 'no'
Beaker::Host::CommandFailure:
  Host 'ubuntu-1804-x64' exited with 8 running:
   wget -O /tmp/puppet.deb http://apt.puppetlabs.com/puppetlabs-release-pc1-bionic.deb
  Last 200 lines of output were:
  	--2018-05-07 12:59:18--  http://apt.puppetlabs.com/puppetlabs-release-pc1-bionic.deb
  	Resolving apt.puppetlabs.com (apt.puppetlabs.com)... 52.222.168.15, 52.222.168.204, 52.222.168.85, ...
  	Connecting to apt.puppetlabs.com (apt.puppetlabs.com)|52.222.168.15|:80... connected.
  	HTTP request sent, awaiting response... 404 Not Found
  	2018-05-07 12:59:19 ERROR 404: Not Found.
```

Also discussed with @traylenator, including his previous PR's changes: https://github.com/voxpupuli/puppet-fail2ban/pull/54 

#### This Pull Request (PR) fixes the following issues

Fixes #36 
Fixes #57

### CODE REVIEW

File: spec/acceptance/class_spec.rb [hunk 1]
```diff
@@ -99,7 +111,7 @@ class { 'fail2ban':
             service_enable => false,
           }
         EOS
-
+
```
Comment (ekohl) [inline spec/acceptance/class_spec.rb]: Could you remove the trailing whitespace here?

File: spec/acceptance/class_spec.rb [hunk 2]
```diff
@@ -13,14 +13,18 @@
...
-  shell('wget http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm')
-  shell('rpm -ivh epel-release-latest-7.noarch.rpm && yum -y install redhat-lsb-core')
 end
 
 # Ensure the ssh log file is created, otherwise the service doesn't start completely
 shell("touch #{ssh_log_file}")
 
+# CentOS6 is really picky, puppet/beaker doesn't see the fact correctly, which is why it's better left as the "default"
+# It's an ugly hack but it works.
+distname = fact('lsbdistcodename')
```
Comment (ekohl) [inline spec/acceptance/class_spec.rb]: This fact should be present if you install `redhat-lsb-core`. Doing these kind of things outside of describe or other blocks really screws up the order and this can be done before the spec helper is done running. That could mean it's applying this to CentOS 7 as well.

File: .travis.yml [hunk 3]
```diff
@@ -27,25 +27,31 @@ matrix:
     env: PUPPET_VERSION="~> 5.0" CHECK=build DEPLOY_TO_FORGE=yes
   - rvm: 2.4.3
     bundler_args: --without development release
-    dist: trusty
+    dist: bionic
```
Comment (ekohl) [inline .travis.yml]: This is from https://github.com/voxpupuli/modulesync_config so if we're going to switch, it needs to happen there.

File: spec/acceptance/nodesets/docker/ubuntu-18.04.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
```
Comment (ekohl) [inline spec/acceptance/nodesets/docker/ubuntu-18.04.yml]: I'd leave this out if it's not used yet.

File: templates/bionic/etc/fail2ban/jail.conf.erb [hunk 5]
```diff
@@ -0,0 +1,864 @@
+#
```
Comment (ekohl) [inline templates/bionic/etc/fail2ban/jail.conf.erb]: I'd keep all Ubuntu changes in a separate PR from this CentOS 6 acceptance testing PR.

**Comment by saibot94** [comment]
Closing as it was done in https://github.com/voxpupuli/puppet-fail2ban/pull/54
